bundles/mapping/mapmodule/service/SelectedFeatureHandler.js

```diff
@@ -0,0 +1,178 @@
+import { WFS_ID_KEY } from '../domain/constants';
+import { getOlStyleForLayer } from '../oskariStyle/generator.ol';
+const getSelectedLayer = layerId => Oskari.getSandbox().findMapLayerFromSelectedMapLayers(layerId);
+
+const SELECTED_STYLE = {
+    inherit: true,
+    effect: 'auto major',
+    stroke: {
+        area: {
+            effect: 'none',
+            color: '#000000',
+            width: 4
+        },
+        width: 3
+    }
+};
+
+export class SelectedFeatureHandler {
```

Comment:
I'd like to think it's a separate concern to track which features form a "selection" and how that selection is reflected on the map. Also I'm not sure if it's good to have the same service trying to dig up the features from OpenLayers sources/layers since it creates a tight coupling with the mapping library. I don't have a good answer how this should be handled but the PR changes both how to select features and track that selection AND also the styling for vector features so it's a lot to take in.